src/rules/rust_style.rs

```diff
@@ -0,0 +1,1922 @@
+          );
+        }
+        if let Some(type_ann) = &getter_sig.type_ann {
+          self.check_ts_type(&type_ann.type_ann);
+        }
+      }
+      TsSetterSignature(setter_sig) => {
+        if let ast_view::Expr::Ident(ident) = setter_sig.key {
+          self.check_ident_snake_cased(
+            ident,
+            IdentToCheck::function(ident.inner),
+          );
+        }
+      }
+      TsIndexSignature(_)
+      | TsCallSignatureDecl(_)
+      | TsConstructSignatureDecl(_) => {}
+    }
+  }
+
+  fn check_pat(&mut self, pat: &ast_view::Pat, ctx: &mut Context) {
+    match pat {
+      ast_view::Pat::Ident(ident) => {
+        //todo: ?ctx always valid?
+        if let Some(v) = ctx.scope().var_by_ident(ident.id) {
+          if let deno_ast::BindingKind::Const = v.kind() {
+            self.check_ident_snake_cased_or_screaming_snake_cased(
+              ident,
+              IdentToCheck::variable_or_constant(ident.id.inner),
+            );
+          } else {
+            self.check_ident_snake_cased(
+              ident,
+              IdentToCheck::variable(ident.id.inner),
+            );
+          }
+        } else {
+          self.check_ident_snake_cased(
+            ident,
+            IdentToCheck::variable(ident.id.inner),
+          );
+        }
+      }
+      ast_view::Pat::Array(ast_view::ArrayPat { elems, .. }) => {
+        for pat in elems.iter().flatten() {
+          self.check_pat(pat, ctx);
+        }
+      }
+      ast_view::Pat::Rest(ast_view::RestPat { ref arg, .. }) => {
+        self.check_pat(arg, ctx);
+      }
+      ast_view::Pat::Object(ast_view::ObjectPat { props, .. }) => {
+        for prop in *props {
+          match prop {
+            ast_view::ObjectPatProp::KeyValue(ast_view::KeyValuePatProp {
+              ref key,
+              ref value,
+              ..
+            }) => match value {
+              ast_view::Pat::Ident(value_ident) => {
+                self.check_ident_snake_cased(
+                  value_ident,
+                  IdentToCheck::object_pat(
+                    &key.string_repr().unwrap_or_else(|| "[KEY]".to_string()),
+                    Some(&value_ident.id.inner),
+                    false,
+                    pat_in_var_declarator(pat.into()),
+                  ),
+                );
+              }
+              ast_view::Pat::Assign(ast_view::AssignPat {
+                left: ast_view::Pat::Ident(value_ident),
+                ..
+              }) => {
+                self.check_ident_snake_cased(
+                  value_ident,
+                  IdentToCheck::object_pat(
+                    &key.string_repr().unwrap_or_else(|| "[KEY]".to_string()),
+                    Some(&value_ident.id.inner),
+                    true,
+                    pat_in_var_declarator(pat.into()),
+                  ),
+                );
+              }
+              _ => {
+                self.check_pat(value, ctx);
```

Comment:
?is `ctx` valid everywhere, because it is passed to nested entities, see todo line 538?